patches/tModLoader/Terraria/ModLoader/SystemLoader.cs

```diff
@@ -37,9 +40,31 @@ internal static void Unload()
 		SystemsByMod.Clear();
 	}
 
-	internal static void ResizeArrays()
+	internal static void ResizeArrays(bool unloading)
 	{
 		RebuildHooks();
+
+		if (unloading) {
+			return;
+		}
+		foreach (var mod in ModLoader.Mods) {
+			var reinitializeTypes = AssemblyManager.GetLoadableTypes(mod.Code)
+				.Where(t => t.GetAttribute<ReinitializeDuringResizeArraysAttribute>() != null)
+				//.Where(t => !t.IsAbstract && !t.ContainsGenericParameters) // static is IsAbstract and IsSealed
```

Comment:
No need to keep these commented out. But we might want to note that `[ReinitializeOnResizeArrays]` can't work on generic classes